app/models/loan.rb

```diff
@@ -149,64 +137,50 @@ def signing_date_long
     I18n.l(self.signing_date, format: :long, default: '')
   end
 
-  # def short_description
-  #   self.translation('ShortDescription')
-  # end
-  # def description
-  #   self.translation('Description')
-  # end
-
   def coop_media(limit=100, images_only=false)
-    get_media('Cooperatives', self.cooperative.try(:id), limit, images_only)
+    organization.get_media(limit: limit, images_only: images_only)
   end
 
   def loan_media(limit=100, images_only=false)
-    get_media('Loans', self.id, limit, images_only)
+    self.get_media(limit: limit, images_only: images_only)
   end
 
   def log_media(limit=100, images_only=false)
     media = []
-    begin
-      self.logs("Date").each do |log|
-        media += log.media(limit - media.count, images_only)
-        return media unless limit > media.count
-      end
-    rescue Mysql2::Error # some logs have invalid dates
+    self.project_logs.find_each do |log|
+      media += log.get_media(limit: limit - media.count, images_only: images_only)
+      return media unless limit > media.count
     end
-    return media
+    media
   end
 
   def featured_pictures(limit=1)
     pics = []
-    coop_pics = get_media('Cooperatives', self.cooperative.try(:id), limit, images_only=true).to_a
+    coop_pics = coop_media(limit, images_only=true).to_a
     # use first coop picture first
     pics << coop_pics.shift if coop_pics.count > 0
     return pics unless limit > pics.count
     # then loan pics
-    pics += get_media('Loans', self.id, limit - pics.count, images_only=true)
+    pics += loan_media(limit - pics.count, images_only=true)
     return pics unless limit > pics.count
     # then log pics
-    pics += self.log_media(limit - pics.count, images_only=true)
+    pics += log_media(limit - pics.count, images_only=true)
     return pics unless limit > pics.count
-    # then remaining coop pics
+  # then remaining coop pics
     pics += coop_pics[0, limit - pics.count]
     return pics
   end
 
   def thumb_path
     if !self.featured_pictures.empty?
-      self.featured_pictures.first.paths[:thumb]
+      self.featured_pictures.first.item.thumb.url
     else "/assets/ww-avatar-watermark.png" end
   end
 
   def ensure_currency
     currency || Currency.find_by(code: 'USD')
   end
```

Comment:
Is this still needed?

Reply:
the `ensure_currency` method is used admin-side in the breakeven tables, so presumably it is still needed